### ISSUE TITLE
Sphinx html_use_smartypants is deprecated as of 1.6 and removed in 1.7

### ISSUE THREAD

**Issue by csabella** [spyder-ide/spyder#6570]
http://www.sphinx-doc.org/en/stable/config.html#confval-smartquotes

This is used in spyder/utils/help/conf.py

**Comment by ccordoba12** [comment]
@csabella, could this be related to issue #6549?

**Comment by csabella** [comment]
@ccordoba12   I don't think if it will fix #6549, but I don't know that for sure.  I can get into the Tutorial on Linux regardless of my version of Sphinx.

I should have included this in the original report.  I got the warning running the test for the help plugin.
```
spyder/plugins/tests/test_help.py::test_no_docs_message
  /home/cheryl/anaconda3/envs/spyder-dev/lib/python3.6/site-packages/sphinx/environment/__init__.py:683: RemovedInSphinx17Warning: html_use_smartypants option is deprecated. Smart quotes are on by default; if you want to disable or customize them, use the smart_quotes option in docutils.conf.
    RemovedInSphinx17Warning)

spyder/plugins/tests/test_help.py::test_no_further_docs_message
  /home/cheryl/anaconda3/envs/spyder-dev/lib/python3.6/site-packages/sphinx/environment/__init__.py:683: RemovedInSphinx17Warning: html_use_smartypants option is deprecated. Smart quotes are on by default; if you want to disable or customize them, use the smart_quotes option in docutils.conf.
    RemovedInSphinx17Warning)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
```
With the patch and on 1.6.3, the warning goes away.  Oddly, without the patch and on Sphinx 1.7.1, the tests run fine.  I guess it just ignores the setting.
